src/polyswarm_api/api.py

```diff
@@ -824,17 +824,54 @@ def event_list(self, **kwargs):
         return resources.Events.list(self, **kwargs).result()
 
     def report_create(self, **kwargs):
-        return resources.ReportTask.create(self, **kwargs).result()
+        nowait = kwargs.pop('nowait', False)
```

Comment:
Hey, yea I didn't want to add much arguments to the `report_create` method that was just released, but nobody is using it yet, so I can add them, moreover it's going to be backward compatible anyway because the way `**kwargs` works. I based my code for the new APIs in some other methods I found in the code that only receive a `**kwargs` argument, but it's a bad practice, and I did not figure out earlier that it was mostly used by `GET` APIs that could receive different arguments for search that's why the "abuse" of `**kwargs` to receive them. 

The "wait_for..." thing it's mostly copy & paste from the scan and sandbox APIs, I understand your point of view about design, but being consistent across the different APIs is also a good design choice, plus it's a nice add-on, after all it's optional and defaulted to false.

I'm going to refactor this method and send it back for review.

Reply:
Just so that you know, we don't have any wait functionality for sandbox tasks, only for scans, and the `wait_for()` method that is used to wait for the scan to finish is used in the cli, not in the api. So, if you want consistency, you can provide the `wait_for_report()` and have it called from the cli, but the way you are doing right now is inconsistent with our other methods. I would really consider moving the wait functionality to the cli.

Reply:
Also, about it being a good design choice, I disagree. The methods in the `api` should be as much as possible a one-to-one association to an endpoint api call, and sintax sugar and convenience should go to the `cli` package. Providing a separate method independent from the CRUD operations for convenience is acceptable, but bundling the convenience in the CRUD operation method and having other endpoints being called by default is not a good design choice.

Reply:
The method from which I basically copy & pasted is indeed from the API, not the CLI: https://github.com/polyswarm/polyswarm-api/blob/3b16adc854805c004b6b2a2e25be328641266e0a/src/polyswarm_api/api.py#L54-L56

Reply:
Yeah, I see, but if you check the code, this method is not called anywhere in the `api`. It is called from the `cli` package.

Reply:
Oh Ok, fair enough, going to use it from the CLI.

Reply:
Done, could you review again please?